src/en/clouds-gce.md

```diff
@@ -0,0 +1,333 @@
+
+```no-highlight
+NAME                                    EMAIL
+Compute Engine Juju service account     juju-gce-sa@juju-gce-123.iam.gserviceaccount.com
+```
+
+We must now give our chosen service account enough permissions so it can do
+what Juju asks of it. The roles of 'Compute Instance Admin (v1)' and 'Compute
+Security Admin' are sufficient:
+
+```bash
+gcloud projects add-iam-policy-binding juju-gce-123 \
+	--member serviceAccount:juju-gce-sa@juju-gce-123.iam.gserviceaccount.com \
+	--role roles/compute.instanceAdmin.v1
+gcloud projects add-iam-policy-binding juju-gce-123 \
+	--member serviceAccount:juju-gce-sa@juju-gce-123.iam.gserviceaccount.com \
+	--role roles/compute.securityAdmin
+```
+
+Permissions can be configured in multiple ways due to the many IAM roles
+available. See upstream document [Compute Engine IAM Roles][gce-iam-roles] for
+details.
+
+Verify the roles now assigned to both your user and your service account:
+
+```bash
+gcloud projects get-iam-policy juju-gce-123
+```
+
+### Gathering credential information
+
+You are now ready to download credentials for your chosen service account.
+Here we've called the download file `juju-gce-sa.json`:
+
+```bash
+gcloud iam service-accounts keys create juju-gce-sa.json \
+	--iam-account=juju-gce-sa@juju-gce-123.iam.gserviceaccount.com
+```
+
+Store this file on the Juju client (e.g. `~/.local/share/juju/juju-gce-sa.json`).
+
+The section [Using environment variables][#using-environment-variables] below
+explains where this data can be stored if you wish to use the
+`autoload-credentials` command to add credentials to Juju.
+
+### Enabling the Compute Engine API
+
+The Compute Engine API needs to be enabled for your project but this requires 
+your billing account to be first linked to your project.
+
+Your billing (credit card) should have been set up when you registered with
+GCE. To see your billing account number:
+
+```bash
+gcloud alpha billing accounts list
+```
+
+Sample output:
+
+```no-highlight
+ACCOUNT_ID            NAME                OPEN  MASTER_ACCOUNT_ID
+01ACD0-B3D759-187641  My Billing Account  True
+```
+
+Use the account number/ID to link your project:
+
+```bash
+gcloud alpha billing projects link juju-gce-123 --billing-account 01ACD0-B3D759-187641
+```
+
+You can now enable the Compute Engine API for your project (this can take a few
+minutes to complete):
+
+```bash
+gcloud services enable compute.googleapis.com --project juju-gce-123
+```
+
+Verify by listing all currently enabled services/APIs:
+
+```bash
+gcloud services list --project juju-gce-123
+```
+
+## Adding credentials
+
+The [Cloud credentials][credentials] page offers a full treatment of credential
+management.
+
+In order to access Google GCE, you will need to add credentials to Juju. This
+can be done in one of three ways (as shown below).
+
+Alternately, you can use your credentials with [Juju as a Service][jaas], where
+charms can be deployed using a web GUI.
```

Comment:
I think JAAS is more than just a GUI, maybe mention you get a free controller too

Reply:
:ok: